packages/code-studio/src/main/SessionUtils.ts

```diff
@@ -15,6 +15,11 @@ export function getWebsocketUrl(): string {
   return `${baseUrl.protocol}//${baseUrl.host}`;
 }
 
+export function isAuthRequired(): boolean {
+  const searchParams = new URLSearchParams(window.location.search);
+  return searchParams.get('authRequired') != null;
+}
```

Comment:
I think we should be more explicit about the auth mechanism here; we're just saying `authRequired`, and then assuming that means we ask the parent window for an authentication token, which will be kind of confusing when we add other authentication methods.

Param like `authProvider=window.opener` or just `authProvider=opener` would make it more explicit, and then instead of calling it `requestAuthToken`, it'd be `requestAuthTokenFromOpener`. (Or `parent` instead of `opener`, your call).

Reply:
Agree. Changed the query param to `authProvider=parent`, request to `requestAuthTokenFromParent`, and response to
```
{ message: 'loginOptions', payload: { type, token }:LoginOptions }
```